pkg/prometheus/promcfg.go

```diff
@@ -2064,6 +2065,16 @@ func (cg *ConfigGenerator) appendScrapeIntervals(slice yaml.MapSlice) yaml.MapSl
 	return slice
 }
 
+func (cg *ConfigGenerator) appendScrapeProtocols(slice yaml.MapSlice) yaml.MapSlice {
+	cpf := cg.prom.GetCommonPrometheusFields()
+
+	if len(cpf.ScrapeProtocols) > 0 {
+		slice = cg.WithMinimumVersion("2.49.0").AppendMapItem(slice, "scrape_protocols", cpf.ScrapeProtocols)
+	}
+
+	return slice
```

Comment:
(nit) a bit more compact
```suggestion
	cpf := cg.prom.GetCommonPrometheusFields()

	if len(cpf.ScrapeProtocols) == 0 {
		return slice
	}

	return cg.WithMinimumVersion("2.49.0").AppendMapItem(slice, "scrape_protocols", cpf.ScrapeProtocols)
```